src/source/ObjcGenerator.scala

```diff
@@ -417,6 +420,44 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
       }
       w.wl
 
+      w.wl("- (NSDictionary *)toDict")
+      w.braced {
+        w.wl("#define _djinni_hide_null_(_o_) ((_o_)?(_o_):([NSNull null]))")
+        w.wl("")
+        w.w(s"return ").nestedN(2) {
+          w.w("@{")
+          w.w("@\"__class_name__\": [self.class description]")
+
+          for (f <- r.fields) {
+            w.w(", @\"")
+            w.w(idObjc.field(f.ident))
+            w.w("\": ")
+
+            w.w("_djinni_hide_null_(")
+
+            f.ty.resolved.base match {
+              case MOptional => w.w(s"self.${idObjc.field(f.ident)}")
+              case t: MPrimitive => w.w(s"@(self.${idObjc.field(f.ident)})")
+              case df: MDef => df.defType match {
+                case DEnum => w.w(s"@(self.${idObjc.field(f.ident)})")
+                case _ => w.w(s"self.${idObjc.field(f.ident)}")
```

Comment:
Is the conversion here shallow?  I'd expect to see some recursion here, in the form of calls to [self.field toDict].  If the intent is to convert this record into a type which might be easily converted to JSON, you'll need to process recursively.  I.e. if one of the fields of RecordA has type RecordB, it's of limited value to just put the RecordB object as a value in a dict, because the result still contains custom records.  I wonder what the target use case is for this, and whether it should really be promising to "normalize" all of the data to only standard NS types (NSDictionary, NSArray, NSNumber).